src/assets/scripts/client/aircraft/AircraftController.js

```diff
@@ -16,31 +23,110 @@ import { GAME_EVENTS } from '../game/GameController';
 // Temporary const declaration here to attach to the window AND use as internal property
 const aircraft = {};
 
-// TODO: this should be renamed to `AircraftCollection`
 /**
  * @class AircraftController
  */
 export default class AircraftController {
     /**
      * @constructor
+     * @for AircraftController
+     * @param aircraftTypeDefinitionList {array<object>}
+     * @param airlineController {AirlineController}
+     * @param navigationLibrary {NavigationLibrary}
      */
-    constructor() {
+    constructor(aircraftTypeDefinitionList, airlineController, navigationLibrary) {
+        if (!_isArray(aircraftTypeDefinitionList) || _isEmpty(aircraftTypeDefinitionList)) {
+            // eslint-disable-next-line max-len
+            throw new TypeError('Invalid aircraftTypeDefinitionList passed to AircraftCollection. Expected and array but ' +
+                `received ${typeof aircraftTypeDefinitionList}`);
+        }
+
+        // TODO: this may need to use instanceof instead, but that may be overly defensive
+        if (!_isObject(airlineController) || !_isObject(navigationLibrary)) {
+            throw new TypeError('Invalid parameters. Expected airlineCollection and navigationLibrary to be defined');
+        }
+
+        /**
+         * Reference to an `AirlineController` instance
+         *
+         * @property airlineController
+         * @type {AirlineController}
+         * @default airlineController
+         */
+        this.airlineController = airlineController;
+
+        /**
+         * Reference to a `NavigationLibrary` instance
+         *
+         * @property navigationLibrary
+         * @type NavigationLibrary
+         * @default navigationLibrary
+         */
+        this.navigationLibrary = navigationLibrary;
+
+        // TODO: rename to `AircraftTypeDefinitionCollection`
+        /**
+         * Reference to an `AircraftCollection` instance
+         *
+         * Provides definitions for all available aircraft types
+         *
+         * @property aircraftCollection
+         * @type {AircraftCollection}
+         */
+        this.aircraftCollection = new AircraftCollection(
+            aircraftTypeDefinitionList,
+            this.airlineController.airlineCollection,
+            this.navigationLibrary
+        );
+
+
         this.aircraft = aircraft;
+        // TODO: replace with aircraftCollection
         this.aircraft.models = {};
+        // TODO: replace with facade to `airlineController`
```

Comment:
This is a design pattern that uses a single method to hide a complex group of additional methods that also get called. (https://sourcemaking.com/design_patterns/facade)

Take a look at some of the stuff that is the `NavigationLibrary` towards the bottom of that file, those are facade methods.  A single class method that is exposed and will, in turn, call a bunch of private methods to get the thing we need so it can be returned.  This principle is used a lot in the `StandardRoute` classes.